File: src/main/java/org/zenbot/timetableupdater/dao/RouteRepository.java
package org.zenbot.timetableupdater.dao;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.zenbot.timetableupdater.domain.BusRoute;

public interface RouteRepository extends MongoRepository<BusRoute, String> {

    BusRoute findByRoutename(String routename);
}

File: src/main/kotlin/org/zenbot/timetableupdater/dao/RouteRepository.kt
package org.zenbot.timetableupdater.dao

import org.springframework.data.repository.CrudRepository
import org.zenbot.timetableupdater.domain.BusRoute

interface RouteRepository : CrudRepository<BusRoute, String> {
    fun findByRoutename(routename: String): BusRoute
}